SpatialGDK/Source/SpatialGDK/Public/Schema/PlayerSpawner.h

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Schema/Component.h"
+#include "SpatialConstants.h"
+#include "Utils/SchemaUtils.h"
+
+#include "Containers/UnrealString.h"
+#include "Engine/EngineBaseTypes.h"
+#include "Engine/GameInstance.h"
+#include "GameFramework/OnlineReplStructs.h"
+#include "Kismet/GameplayStatics.h"
+#include "UObject/CoreNet.h"
+
+#include <WorkerSDK/improbable/c_schema.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+namespace SpatialGDK
+{
+
+struct SpawnPlayerRequest
+{
+	FURL LoginURL;
+	FUniqueNetIdRepl UniqueId;
+	FName OnlinePlatformName;
+	bool bIsSimulatedPlayer;
+};
+
+struct PlayerSpawner : Component
+{
+	static const Worker_ComponentId ComponentId = SpatialConstants::PLAYER_SPAWNER_COMPONENT_ID;
+
+	PlayerSpawner() = default;
+
+	static Worker_CommandRequest CreatePlayerSpawnRequest(SpawnPlayerRequest& SpawnRequest)
+	{
+		Worker_CommandRequest CommandRequest = {};
+		CommandRequest.component_id = SpatialConstants::PLAYER_SPAWNER_COMPONENT_ID;
+		CommandRequest.command_index = SpatialConstants::PLAYER_SPAWNER_SPAWN_PLAYER_COMMAND_ID;
+		CommandRequest.schema_type = Schema_CreateCommandRequest();
+
+		Schema_Object* RequestFields = Schema_GetCommandRequestObject(CommandRequest.schema_type);
+		AddSpawnPlayerData(RequestFields, SpawnRequest);
+
+		return CommandRequest;
+	}
+
+	static Worker_CommandResponse CreatePlayerSpawnResponse()
+	{
+		Worker_CommandResponse CommandResponse = {};
+		CommandResponse.component_id = SpatialConstants::PLAYER_SPAWNER_COMPONENT_ID;
+		CommandResponse.command_index = SpatialConstants::PLAYER_SPAWNER_SPAWN_PLAYER_COMMAND_ID;
+		CommandResponse.schema_type = Schema_CreateCommandResponse();
+		return CommandResponse;
+	}
+
+	static void AddSpawnPlayerData(Schema_Object* RequestObject, SpawnPlayerRequest& SpawnRequest)
```

Comment:
```suggestion
	static void AddSpawnPlayerData(Schema_Object* RequestObject, const SpawnPlayerRequest& SpawnRequest)
```

Reply:
I don't think we can since the << operator below doesn't take const values on rhs

Reply:
Ah, cause it doesn't know if it's reading or writing at that point. Fine.